CLI Overview

jwitz opened the pull request.

paolaperaza commented on astro/cli/overview.md:
```diff
@@ -0,0 +1,26 @@
+description: Learn about every command that you can run with the Astro CLI.
+---
+
+## Build, Run, and Observe Airflow Locally
```
```suggestion
## Get Started Locally
```

I like the BRO framework, but might be a little bold in the context of the CLI here

paolaperaza commented on astro/cli/overview.md:
```diff
@@ -0,0 +1,26 @@
+
+## Build, Run, and Observe Airflow Locally
+
+The open source Astro CLI makes it easy to develop an Airflow project and run it on your local machine. It automatically manages Airflow resources and components so that you can focus on development, not configuration.
```
```suggestion
The Astro CLI is an open source command line interface for data orchestration. It's the easiest way to get started with [Apache Airflow](https://airflow.apache.org/) and can be used with all Astronomer products.

As a complement to Apache Airflow, the Astro CLI manages the creation of all Airflow components required to run DAGs locally. It is also built with Astro Runtime, a Docker image maintained by Astronomer that offers timely access to both the latest Airflow versions as well as differentiating features that center on performance, reliability, and usability.

The Astro CLI is open source and built for data practitioners everywhere. The binary is maintained in the public [Astro CLI GitHub repository](https://github.com/astronomer/astro-cli), where pull requests and GitHub issues are welcome.
```

paolaperaza commented on astro/cli/overview.md:
```diff
@@ -0,0 +1,26 @@
+## Build, Run, and Observe Airflow Locally
+
+The open source Astro CLI makes it easy to develop an Airflow project and run it on your local machine. It automatically manages Airflow resources and components so that you can focus on development, not configuration.
+
```
```suggestion
## Features

The Astro CLI includes:

- A built-in Astro project directory that includes all the files required to run Airflow, including dedicated folders for your DAGs, packages, and unit tests.
- A set of commands for local development that make it easy to start your local environment, apply code changes, and view logs.
- A set of example pytests and example DAGs that showcase important Airflow best practices and can help your team learn quickly and identify errors in your DAGs ahead of time.
- Easy and secure browser-based authentication for Astro and Astronomer Software.
- A robust set of commands to match functionality in the Cloud UI, including commands to create a Deployment and modify environment variables.
- Support for Deployment API keys, which allows you to automate Astro CLI commands as part of CI/CD workflows.

To stay up to date on the latest features, see [Astro CLI Release Notes](https://github.com/astronomer/docs/pull/730/cli-release-notes.md).

```

paolaperaza commented on astro/cli/overview.md:
```diff
@@ -0,0 +1,26 @@
+
+The open source Astro CLI makes it easy to develop an Airflow project and run it on your local machine. It automatically manages Airflow resources and components so that you can focus on development, not configuration.
+
+## Manage Astro From Your Terminal
+
+The Astro CLI also functions as a core tool for deploying Airflow at scale with [Astro](https://www.astronomer.io/product/). Using the Astro CLI, you can:
+
+- Manage user permissions in Astro Workspaces and Deployments.
+- Automatically deploy code to Astro via CI/CD.
+- Access logs for Airflow components running in your cloud.
```
I'd delete this TBH... I think we tow the line of Astro vs. local in the features section decently well?

paolaperaza commented on astro/cli/overview.md:
```diff
@@ -0,0 +1,26 @@
+- Automatically deploy code to Astro via CI/CD.
+- Access logs for Airflow components running in your cloud.
+
+## Quickstart
```
```suggestion
## Get Started
```

paolaperaza replied:
I struggle with Quickstart vs. Get Started but can get over it!!

paolaperaza commented on astro/cli/overview.md:
```diff
@@ -0,0 +1,26 @@
+
+## Quickstart
+
+Follow the [Astro CLI Quickstart](cli-quickstart.md) to install the CLI and run your first locally hosted Airflow project.
```
```suggestion
To get started locally, follow the [Astro CLI Quickstart](cli-quickstart.md) for end-to-end instructions.

To install or upgrade the Astro CLI, see [Configure the CLI](install-cli.md).
```

paolaperaza reviewed (commented):
@jwitz Some thoughts here! Changed it up a good amount, LMK what you think

jwitz commented:
@paolaperaza I like the new Features section! Returning this back for review with a few high level changes:

- I changed the first title to be a little more bold and to provide more value add up front
- I removed details about Runtime. I think it will end up confusing CLI newbies that don't necessarily want to worry about what image they're running. Maybe we can include something about Runtime in the quickstart? The way we frame it here, it was unclear whether Runtime was a part of the CLI or its own thing. 
- I shifted the quickstart up to before the features list, because I think that's still the best demonstration of features that we have.

paolaperaza commented on astro/cli/overview.md:
```diff
@@ -0,0 +1,35 @@
+
+## Run Airflow Locally
+
+The Astro CLI is an open source command line interface for data orchestration. It's the easiest way to get started with [Apache Airflow](https://airflow.apache.org/) and can be used with all Astronomer products.
```
```suggestion
The Astro CLI is the command line interface for data orchestration. It's the easiest way to get started with [Apache Airflow](https://airflow.apache.org/) and can be used with all Astronomer products.
```

I guess felt repetitive with same start on sentence below

paolaperaza commented on astro/cli/overview.md:
```diff
@@ -0,0 +1,35 @@
+
+## Features
+
+As a complement to Apache Airflow, the Astro CLI manages the creation of all Airflow components required to run DAGs locally.
```
```suggestion
```

I think if we include this as an overarching statement that leads "all features" it's too narrow, and we kinda already stated its purpose in the intro. I'd just delete here or have it above?

paolaperaza commented on astro/cli/overview.md:
```diff
@@ -0,0 +1,35 @@
+
+Specifically, the Astro CLI includes:
+
+- A built-in project directory that includes all the files required to run Airflow, including dedicated folders for your DAGs, packages, and unit tests.
```
```suggestion
- A built-in Astro project directory that includes all the files required to run Airflow, including dedicated folders for your DAGs, packages, and unit tests.
```

paolaperaza commented on astro/cli/overview.md:
```diff
@@ -0,0 +1,35 @@
+Specifically, the Astro CLI includes:
+
+- A built-in project directory that includes all the files required to run Airflow, including dedicated folders for your DAGs, packages, and unit tests.
+- Commands to enhance the local development experience. Using the CLI, you can run a local Airfow environment, apply code changes, and view system component logs.
```
```suggestion
- Commands to enhance the local development experience. Using the CLI, you can run a local Airfow environment, apply code changes, and view logs for all Airflow components.
```

Not so much "system" components idt

paolaperaza reviewed (approved):
Yay, final minor comments but LGTM @jwitz
